excel_import_export/models/common.py

```diff
@@ -0,0 +1,372 @@
+        'fill': {
+            'red': PatternFill("solid", fgColor="FF0000"),
+            'grey': PatternFill("solid", fgColor="DDDDDD"),
+            'yellow': PatternFill("solid", fgColor="FFFCB7"),
+            'blue': PatternFill("solid", fgColor="9BF3FF"),
+            'green': PatternFill("solid", fgColor="B0FF99"),
+        },
+        'align': {
+            'left': Alignment(horizontal='left'),
+            'center': Alignment(horizontal='center'),
+            'right': Alignment(horizontal='right'),
+        },
+        'number_format': {
+            'number': '#,##0.00',
+            'date': 'dd/mm/yyyy',
+            'datestamp': 'yyyy-mm-dd',
+            'text': '@',
+            'percent': '0.00%',
+        },
+    }
+    formats = field_format.split(';')
+    for f in formats:
+        (key, value) = f.split('=')
+        if key not in avail_format.keys():
+            raise ValidationError(_('Invalid format type %s' % key))
+        if value.lower() not in avail_format[key].keys():
+            raise ValidationError(
+                _('Invalid value %s for format type %s' % (value, key)))
+        cell_format = avail_format[key][value]
+        if key == 'font':
+            field.font = cell_format
+        if key == 'fill':
+            field.fill = cell_format
+        if key == 'align':
+            field.alignment = cell_format
+        if key == 'number_format':
+            if value == 'text':
+                try:
+                    # In case value can't be encoded as utf, we do normal str()
+                    field.value = field.value.encode('utf-8')
+                except Exception:
+                    field.value = str(field.value)
+            field.number_format = cell_format
+
+
+def get_line_max(line_field):
+    """ i.e., line_field = line_ids[100], max = 100 else 0 """
+    if line_field and '[' in line_field and ']' in line_field:
+        i = line_field.index('[')
+        j = line_field.index(']')
+        max_str = line_field[i + 1:j]
+        try:
+            if len(max_str) > 0:
+                return (line_field[:i], int(max_str))
+            else:
+                return (line_field, False)
+        except Exception:
+            return (line_field, False)
+    return (line_field, False)
+
+
+def get_groupby(line_field):
+    """i.e., line_field = line_ids["a_id, b_id"], groupby = ["a_id", "b_id"]"""
+    if line_field and '[' in line_field and ']' in line_field:
+        i = line_field.index('[')
+        j = line_field.index(']')
+        groupby = literal_eval(line_field[i:j+1])
+        return groupby
+    return False
+
+
+def split_row_col(pos):
+    match = re.match(r"([a-z]+)([0-9]+)", pos, re.I)
+    if not match:
+        raise ValidationError(_('Position %s is not valid') % pos)
+    col, row = match.groups()
+    return col, int(row)
+
+
+def openpyxl_get_sheet_by_name(book, name):
+    """ Get sheet by name for openpyxl """
+    i = 0
+    for sheetname in book.sheetnames:
```

Comment:
Because it loops by sheetnames, it will never have index error, while another one is need to loop by itertools.count() index. And it is possible that, there is no index.